I wanted to upstream an updated done myself and used for one migration (imap-to-imap) when I noticed your PR. I tried to review / merge it with my changes but I'm unable to get it building on "x86_64-linux".

`
bsMF42 | "/nix/store/p743jnywg5z2axcrdcfyjjqkgdlpjy8r-perl-5.36.0/bin/perl" -Mblib=.. run_with_inc.pl ./boot -q -B -Oparl
bsMF42 | ./boot: extraction of /build/Tr9PXHN15u/par-6e6978626c64/cache-10e91eb0fcd0ac81118024db75ad927b8fd58d98//nix/store/bzd91shky9j9d43girrrj6vmqlw7x9m8-glibc-2.35-163/lib/ld-linux-x86-64.so.2 failed (errno=2)
bsMF42 | system(./boot -I/nix/store/qsrzwhdn2jwpnx9mv89n8496dxzmk9q8-perl5.36.0-PAR-Packer-1.056/lib/perl5/site_perl -I/nix/store/g0zhf66c9p7lgsb2bii192ag53qsdb2a-perl5.36.0-PAR-Dist-0.51/lib/perl5/site_perl/5.36.0/x86_64-linux-thread-multi -I/nix/store/g0zhf66c9p7lgsb2bii192ag53qsdb2a-perl5.36.0-PAR-Dist-0.51/lib/perl5/site_perl/5.36.0 -I/nix/store/g0zhf66c9p7lgsb2bii192ag53qsdb2a-perl5.36.0-PAR-Dist-0.51/lib/perl5/site_perl -I/nix/store/9b5i6l3ryi191qjml9n2xkrd28fagp7r-perl5.36.0-PAR-1.017/lib/perl5/site_perl/5.36.0/x86_64-linux-thread-multi -I/nix/store/9b5i6l3ryi191qjml9n2xkrd28fagp7r-perl5.36.0-PAR-1.017/lib/perl5/site_perl/5.36.0 -I/nix/store/9b5i6l3ryi191qjml9n2xkrd28fagp7r-perl5.36.0-PAR-1.017/lib/perl5/site_perl -I/nix/store/gz31mqagnqbz261f9cj59l538zzg8yaz-perl5.36.0-Module-ScanDeps-1.29/lib/perl5/site_perl/5.36.0/x86_64-linux-thread-multi -I/nix/store/gz31mqagnqbz261f9cj59l538zzg8yaz-perl5.36.0-Module-ScanDeps-1.29/lib/perl5/site_perl/5.36.0 -I/nix/store/gz31mqagnqbz261f9cj59l538zzg8yaz-perl5.36.0-Module-ScanDeps-1.29/lib/perl5/site_perl -I/nix/store/rpp42zmk6sq5hzgz9z9xy0x6ab834sqd-perl5.36.0-IPC-Run3-0.048/lib/perl5/site_perl/5.36.0/x86_64-linux-thread-multi -I/nix/store/rpp42zmk6sq5hzgz9z9xy0x6ab834sqd-perl5.36.0-IPC-Run3-0.048/lib/perl5/site_perl/5.36.0 -I/nix/store/rpp42zmk6sq5hzgz9z9xy0x6ab834sqd-perl5.36.0-IPC-Run3-0.048/lib/perl5/site_perl -I/nix/store/gf9kfkssspikw04gfhkj51lb20rvdpsd-perl5.36.0-Getopt-ArgvFile-1.11/lib/perl5/site_perl/5.36.0/x86_64-linux-thread-multi -I/nix/store/gf9kfkssspikw04gfhkj51lb20rvdpsd-perl5.36.0-Getopt-ArgvFile-1.11/lib/perl5/site_perl/5.36.0 -I/nix/store/gf9kfkssspikw04gfhkj51lb20rvdpsd-perl5.36.0-Getopt-ArgvFile-1.11/lib/perl5/site_perl -I/nix/store/bsy7n44y14fjd823bkx0gmycbm08j2x3-perl5.36.0-Archive-Zip-1.68/lib/perl5/site_perl/5.36.0/x86_64-linux-thread-multi -I/nix/store/bsy7n44y14fjd823bkx0gmycbm08j2x3-perl5.36.0-Archive-Zip-1.68/lib/perl5/site_perl/5.36.0 -I/nix/store/bsy7n44y14fjd823bkx0gmycbm08j2x3-perl5.36.0-Archive-Zip-1.68/lib/perl5/site_perl -I/nix/store/p743jnywg5z2axcrdcfyjjqkgdlpjy8r-perl-5.36.0/lib/perl5/site_perl/5.36.0/x86_64-linux-thread-multi -I/nix/store/p743jnywg5z2axcrdcfyjjqkgdlpjy8r-perl-5.36.0/lib/perl5/site_perl/5.36.0 -I/nix/store/p743jnywg5z2axcrdcfyjjqkgdlpjy8r-perl-5.36.0/lib/perl5/site_perl -I../blib/arch -I../blib/lib -I/nix/store/p743jnywg5z2axcrdcfyjjqkgdlpjy8r-perl-5.36.0/lib/perl5/site_perl/5.36.0/x86_64-linux-thread-multi -I/nix/store/p743jnywg5z2axcrdcfyjjqkgdlpjy8r-perl-5.36.0/lib/perl5/site_perl/5.36.0 -I/nix/store/p743jnywg5z2axcrdcfyjjqkgdlpjy8r-perl-5.36.0/lib/perl5/site_perl -I/nix/store/bsy7n44y14fjd823bkx0gmycbm08j2x3-perl5.36.0-Archive-Zip-1.68/lib/perl5/site_perl/5.36.0/x86_64-linux-thread-multi -I/nix/store/bsy7n44y14fjd823bkx0gmycbm08j2x3-perl5.36.0-Archive-Zip-1.68/lib/perl5/site_perl/5.36.0 -I/nix/store/bsy7n44y14fjd823bkx0gmycbm08j2x3-perl5.36.0-Archive-Zip-1.68/lib/perl5/site_perl -I/nix/store/gf9kfkssspikw04gfhkj51lb20rvdpsd-perl5.36.0-Getopt-ArgvFile-1.11/lib/perl5/site_perl/5.36.0/x86_64-linux-thread-multi -I/nix/store/gf9kfkssspikw04gfhkj51lb20rvdpsd-perl5.36.0-Getopt-ArgvFile-1.11/lib/perl5/site_perl/5.36.0 -I/nix/store/gf9kfkssspikw04gfhkj51lb20rvdpsd-perl5.36.0-Getopt-ArgvFile-1.11/lib/perl5/site_perl -I/nix/store/rpp42zmk6sq5hzgz9z9xy0x6ab834sqd-perl5.36.0-IPC-Run3-0.048/lib/perl5/site_perl/5.36.0/x86_64-linux-thread-multi -I/nix/store/rpp42zmk6sq5hzgz9z9xy0x6ab834sqd-perl5.36.0-IPC-Run3-0.048/lib/perl5/site_perl/5.36.0 -I/nix/store/rpp42zmk6sq5hzgz9z9xy0x6ab834sqd-perl5.36.0-IPC-Run3-0.048/lib/perl5/site_perl -I/nix/store/gz31mqagnqbz261f9cj59l538zzg8yaz-perl5.36.0-Module-ScanDeps-1.29/lib/perl5/site_perl/5.36.0/x86_64-linux-thread-multi -I/nix/store/gz31mqagnqbz261f9cj59l538zzg8yaz-perl5.36.0-Module-ScanDeps-1.29/lib/perl5/site_perl/5.36.0 -I/nix/store/gz31mqagnqbz261f9cj59l538zzg8yaz-perl5.36.0-Module-ScanDeps-1.29/lib/perl5/site_perl -I/nix/store/9b5i6l3ryi191qjml9n2xkrd28fagp7r-perl5.36.0-PAR-1.017/lib/perl5/site_perl/5.36.0/x86_64-linux-thread-multi -I/nix/store/9b5i6l3ryi191qjml9n2xkrd28fagp7r-perl5.36.0-PAR-1.017/lib/perl5/site_perl/5.36.0 -I/nix/store/9b5i6l3ryi191qjml9n2xkrd28fagp7r-perl5.36.0-PAR-1.017/lib/perl5/site_perl -I/nix/store/g0zhf66c9p7lgsb2bii192ag53qsdb2a-perl5.36.0-PAR-Dist-0.51/lib/perl5/site_perl/5.36.0/x86_64-linux-thread-multi -I/nix/store/g0zhf66c9p7lgsb2bii192ag53qsdb2a-perl5.36.0-PAR-Dist-0.51/lib/perl5/site_perl/5.36.0 -I/nix/store/g0zhf66c9p7lgsb2bii192ag53qsdb2a-perl5.36.0-PAR-Dist-0.51/lib/perl5/site_perl -I/nix/store/qsrzwhdn2jwpnx9mv89n8496dxzmk9q8-perl5.36.0-PAR-Packer-1.056/lib/perl5/site_perl -I/nix/store/p743jnywg5z2axcrdcfyjjqkgdlpjy8r-perl-5.36.0/lib/perl5/site_perl/5.36.0/x86_64-linux-thread-multi -I/nix/store/p743jnywg5z2axcrdcfyjjqkgdlpjy8r-perl-5.36.0/lib/perl5/site_perl/5.36.0 -I/nix/store/p743jnywg5z2axcrdcfyjjqkgdlpjy8r-perl-5.36.0/lib/perl5/5.36.0/x86_64-linux-thread-multi -I/nix/store/p743jnywg5z2axcrdcfyjjqkgdlpjy8r-perl-5.36.0/lib/perl5/5.36.0 -I. -q -B -Oparl) failed: $?=65280
bsMF42 | make[1]: *** [Makefile:890: parl] Error 255
bsMF42 | make[1]: Leaving directory '/build/PAR-Packer-1.056/myldr'
bsMF42 | make: *** [Makefile:573: subdirs] Error 2
bsMF42 | error: builder for '/nix/store/qxx1ciyajasx28fbsab24qcc7pp7wywb-perl5.36.0-PAR-Packer-1.056.drv' failed with exit code 2
`
(Full build log: https://privatebin.net/?89d85db95939b1b8#FSGv4JZ9HjuboBRh3gg2ATLao4RN9uj8XKUXCMpwDnTv)

I used a similar changes except the PAR*-dependencies, are they optional or what are the required for? Do you have an idea how to solve this? (after that we could update to 227)

--- a/pkgs/tools/networking/imapsync/default.nix
+++ b/pkgs/tools/networking/imapsync/default.nix
@@ -1,16 +1,25 @@
-{lib, stdenv, makeWrapper, fetchurl, perl, openssl, perlPackages }:
+{ lib
+, fetchFromGitHub
+, makeWrapper
+, perl
+, perlPackages
+, stdenv
+}:
 
 stdenv.mkDerivation rec {
   pname = "imapsync";
-  version = "1.727";
+  version = "2.200";
 
-  src = fetchurl {
-    url = "https://releases.pagure.org/imapsync/imapsync-${version}.tgz";
-    sha256 = "1axacjw2wyaphczfw3kfmi5cl83fyr8nb207nks40fxkbs8q5dlr";
+  src = fetchFromGitHub {
+    owner = "imapsync";
+    repo = "imapsync";
+    rev = "imapsync-${version}";
+    sha256 = "sha256-EM8nT9v6ZHpEG33/ibPqvfgFoAF36nhq6Y+RUWK4vR0=";
   };
 
-  patchPhase = ''
+  postPatch = ''
     sed -i -e s@/usr@$out@ Makefile
+    substituteInPlace INSTALL.d/prerequisites_imapsync --replace "PAR::Packer" ""
   '';
 
   postInstall = ''
@@ -19,18 +28,44 @@ stdenv.mkDerivation rec {
 
   nativeBuildInputs = [ makeWrapper ];
 
-  buildInputs = with perlPackages; [ perl openssl MailIMAPClient TermReadKey
-    IOSocketSSL DigestHMAC URI FileCopyRecursive IOTee UnicodeString
-    DataUniqid JSONWebToken TestMockGuard LWP CryptOpenSSLRSA
-    LWPProtocolHttps Readonly TestPod TestMockObject ParseRecDescent
-    IOSocketInet6 NTLM
+  buildInputs = with perlPackages; [
+    Appcpanminus
+    CGI
+    CryptOpenSSLRSA
+    DataUniqid
+    DistCheckConflicts
+    EncodeIMAPUTF7
+    FileCopyRecursive
+    FileTail
+    IOSocketInet6
+    IOTee
+    JSONWebToken
+    LWP
+    MailIMAPClient
+    ModuleImplementation
+    ModuleScanDeps
+    NTLM
+    PackageStash
+    PackageStashXS
+    Readonly
+    RegexpCommon
+    SysMemInfo
+    TermReadKey
+    TestDeep
+    TestFatal
+    TestMockGuard
+    TestMockObject
+    TestPod
+    TestRequires
+    UnicodeString
+    perl
   ];
 
   meta = with lib; {
-    homepage = "http://www.linux-france.org/prj/imapsync/";
     description = "Mail folder synchronizer between IMAP servers";
-    license = licenses.gpl2Plus;
-    platforms = platforms.linux;
+    homepage = "https://imapsync.lamiral.info/";
+    license = licenses.wtfpl;
     maintainers = with maintainers; [ pSub ];
+    platforms = platforms.unix;
   };
 }

--- a/pkgs/top-level/perl-packages.nix
+++ b/pkgs/top-level/perl-packages.nix
@@ -8307,6 +8307,20 @@ let
     };
   };
 
+  EncodeIMAPUTF7 = buildPerlPackage {
+    pname = "Encode-IMAPUTF7";
+    version = "1.05";
+    src = fetchurl {
+      url = "mirror://cpan/authors/id/P/PM/PMAKHOLM/Encode-IMAPUTF7-1.05.tar.gz";
+      hash = "sha256-RwMF3cN0g8/o08FtE3cKKAEfYAv1V6y4w+B3OZl8N+E=";
+    };
+    checkInputs = [ TestNoWarnings ];
+    meta = {
+      description = "IMAP modified UTF-7 encoding";
+      license = with lib.licenses; [ artistic1 gpl1Plus ];
+    };
+  };
+
   EncodeJIS2K = buildPerlPackage {
     pname = "Encode-JIS2K";
     version = "0.03";

